src/natcap/invest/hydropower/hydropower_water_yield.py

```diff
@@ -415,15 +415,6 @@ def execute(args):
     else:
         demand_lucodes = None
 
-    valid_lulc_txt_path = os.path.join(intermediate_dir, 'valid_lulc_values.txt')
-    check_missing_lucodes_task = graph.add_task(
-        _check_missing_lucodes,
-        args=(clipped_lulc_path, demand_lucodes,
-              bio_lucodes, valid_lulc_txt_path),
-        target_path_list=[valid_lulc_txt_path],
-        dependent_task_list=[align_raster_stack_task],
-        task_name='check_missing_lucodes')
-
```

Comment:
I'm speculating a bit here because I don't have a great memory of it, but I think this task & function was probably designed with two goals in mind. 1) give a good message about missing lucodes, and 2) check for the missing codes as early as possible in execution. The new `utils.reclassify_raster` certainly covers the first purpose, but is there still good reason to keep this around for the second purpose?

Reply:
@davemfish this is a good question and one that I went back and forth on in a few ways. One of my early designs was to have ALL models do a validation-like (not actually in validation) check towards the beginning of execution for table and raster compliance. If memory serves me, @phargogh and I discussed this and I can't remember exactly what we concluded. But, I think ultimately we decided that because of the model contract perhaps doing these extra checks was over designing. On top of that adding these additional computations would be a performance tradeoff when dealing with larger raster inputs. I think that having most of our calls in Tasks helps with this problem as well, since hopefully given a missing value error, subsequent runs can more quickly get back to executing the part of code that failed. 

The more I think about this problem, the more it makes sense to me that it should be a Validation problem or handled as an exception like currently being done. I think I steered away from a Validation implementation for the time being out of concern that a large raster input or table could hamper performance and the user interaction experience with the UI.

Would love to hear thoughts on my ramblings!

Reply:
@dcdenu4 thanks for your thoughts! I think you make a good point that these extra validation-like checks could be over-designing, but I would bet that this check came as a response to reports of failures very late in execution. Anyway now that Taskgraph is in place I agree that this may no longer be a real problem, if it ever was. So purging this sounds good to me.

Reply:
I don't remember everything we talked about, but everything you mention @dcdenu4 sound like the right concerns here!  The required extra read-through of a potentially large raster input sounds like something that could be particularly troublesome since we'll already be needing to check those values as part of the reclassification.